remotes/docker/resolver.go

```diff
@@ -96,6 +96,11 @@ type ResolverOptions struct {
 	// since the registry does not have upload tracking and the existing
 	// mechanism for getting blob upload status is expensive.
 	Tracker StatusTracker
+
+	// OriginProvider returns a slice of refspecs where the Pusher may find candidates for mounting
+	// instead of pushing from local store.
+	// When the Pusher succeeds by mounting, it returns an AlreadyExists error
+	OriginProvider func(ocispec.Descriptor) []reference.Spec
```

Comment:
It's an `ErrAlreadyExists` (`Err` prefix).

Like I commented above, I think this is worth repeating on the godoc for `Push` too.